From 717778f89d7186d086310cbb754180117657cc88 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Wed, 07 Sep 2022 09:49:42 +0300
Subject: [PATCH] [SW-0] gaudi3: pldm: use simulated clk frequency

So far a real world clock equivalent was used for the TimeStamp
counter and uart. This has some significant disadvantages:
1. the real world clk freq on pldm differs from image to image, and
   also from one pldm setup to another.
2. using values below 1MHz is not useful, because the division
   calculation when converting to usec will always return 0.
3. when measuring simulation domain timing, using real world
   equivalent will doesn't help and the simulation clock value should
   be used.

for those reasons, the clock value for pldm is set to simulation
domain frequency value of 250Mhz, which is usually used for PCI
PLL DIV3 output on pldm tests.

Change-Id: I099dd58b314303e568b6cdc63041d824e6e210c6
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index eda3259..031b2eb 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -11,7 +11,7 @@
 #include <pll.h>
 #include "hl.h"
 
-#define SYS_CLOCK_FREQUENCY	21300ULL /* actual freq of clk_250mhz */
+#define SYS_CLOCK_FREQUENCY	250000000ULL /* PCI PLL DIV3 SIMULATION CLK */
 
 #define UART_CLOCK_FREQUENCY	SYS_CLOCK_FREQUENCY
 #define PLDM_UART_BAUDRATE	300
